Bad error message in method resolution

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
struct Foo(i32);

impl Foo {
    fn bar(self) -> i32 {
        self.0 + 1
    }
}

fn main() {
    let a = Foo(123);
    a.baz();
}
```

I expected to see this happen:

```
error[E0599]: no method named `baz` found for struct `Foo` in the current scope
```

Instead, this happened:

```
<source>:11:5: error: failed to resolve the PathExprSegment to any item
   11 |     a.baz();
      |     ^
<source>:11:5: error: failed to type resolve expression
```

### Meta

- What version of Rust GCC were you using, git sha if possible. 69f6be3ee483c9895b4b5187a44b3e1c8be2ba63

--- a/gcc/rust/typecheck/rust-hir-type-check-expr.h
+++ b/gcc/rust/typecheck/rust-hir-type-check-expr.h
@@ -299,17 +299,10 @@ public:
       = MethodResolution::Select (candidates, receiver_tyty, adjustments);
     if (resolved_candidate == nullptr)
       {
-	if (candidates.size () > 1)
-	  {
-	    // not sure if this is the correct error here
-	    ReportMultipleCandidateError::Report (
-	      candidates, expr.get_method_name ().get_segment (),
-	      expr.get_method_name ().get_locus ());
-	  }
-	else
-	  {
-	    rust_error_at (expr.get_locus (), "failed to resolve method");
-	  }
+	rust_error_at (
+	  expr.get_method_name ().get_locus (),
+	  "failed to resolve method for %<%s%>",
+	  expr.get_method_name ().get_segment ().as_string ().c_str ());
 	return;
       }
 

--- a/gcc/testsuite/rust/compile/method1.rs
+++ b/gcc/testsuite/rust/compile/method1.rs
@@ -8,6 +8,6 @@ pub fn main() {
     a = Foo(123);
 
     a.test();
-    // { dg-error "failed to resolve method" "" { target *-*-* } .-1 }
+    // { dg-error "failed to resolve method for .test." "" { target *-*-* } .-1 }
     // { dg-error {failed to type resolve expression} "" { target *-*-* } .-2 }
 }

